src/VideoModal.tsx

```diff
@@ -71,6 +73,40 @@ const VideoModal: React.FC<VideoModalProps> = ({
     }
   }, [selectedVideo, viamClient]);
 
+  const handleShare = () => {
+    if (!selectedVideo) {
+      return;
+    }
+
+    navigator.clipboard.writeText(videoPageURL);
+
+    setVideoShared(true);
+    setTimeout(() => {
+      setVideoShared(false);
+    }, 2000);
+  };
+
+  const handleShareFromCurrentLocation = () => {
+    if (!selectedVideo) {
+      return;
+    }
+
+    const videoPlayer = document.getElementById('video-player') as HTMLVideoElement;
+    if (!videoPlayer) {
+      return;
+    }
+
+    const currentTime = videoPlayer.currentTime;
+
+    const url = `${videoPageURL}&loc=${currentTime}`;
+    navigator.clipboard.writeText(url);
+
+    setVideoSharedFromCurrentLocation(true);
+    setTimeout(() => {
+      setVideoSharedFromCurrentLocation(false);
+    }, 2000);
+  };
+
```

Comment:
are we repeating logic on video detail page? 

also the share links seem to be the same on [video detail page](https://sanding-monitoring-web-app-staging_katie-viam.viamapplications.com/machine/njsanding2-main.tbr1ap8x6t.viam.cloud#/videos/S9RTWzxHCT2RfuyijVC3ifmGcIIod80Hzni7KDYLYOIZcdSGuam70wMZs5a78tRf?name=%2Froot%2F.viam%2Fcapture%2Fvideo-upload%2Fvideo-store-Amcrest-IP5MT1179EWAIV3-AMC10815DB311E510C-url0%2Fvideo-store-Amcrest-IP5MT1179EWAIV3-AMC10815DB311E510C-url0_2025-12-15_17-42-33_956beab8a43a5963c79e3900b61cab7eGeneratingMesh.mp4) vs different on modal